app/controllers/application_controller.rb

```diff
@@ -36,10 +36,10 @@ def csp_headers
   end
 
   def confirm_admin_user
-    redirect_to root_url unless current_user.admin?
+    redirect_to root_url unless current_user.role_admin?
   end
 
   def confirm_user_has_data_access
-    redirect_to root_url unless (current_user.admin? || current_user.data_volunteer?)
+    redirect_to root_url unless (current_user.role_admin? || current_user.role_data_volunteer?)
```

Comment:
Just went with docs, i like it because it actually shows the parent relationship.